chalk-solve/src/clauses/super_traits.rs

```diff
@@ -0,0 +1,130 @@
+use rustc_hash::FxHashSet;
+
+use super::builder::ClauseBuilder;
+use crate::RustIrDatabase;
+use chalk_ir::{
+    fold::shift::Shift, interner::Interner, Binders, BoundVar, DebruijnIndex, TraitId, TraitRef,
+    WhereClause,
+};
+
+pub(super) fn push_where_clause_binders<I: Interner>(
```

Comment:
There's something that bugs me about this function. I'm not quite sure why exactly. I *think* this is too "generalized"; i.e it's a bit unclear what this is supposed to be used for. (Maybe documentation here would help. Maybe it's just poorly named? Not sure.)

Can we just inline this into the two places it's used.